cwm_worker_tests/upload_download_test.py

```diff
@@ -95,78 +109,97 @@ def run(self):
-            }
-
         except S3Error as e:
             print(f'MinIO Exception: {e}')
         except Exception as e:
             print(f'Exception: {e}')
 
 
-def download(endpoint, access_key, secret_key, bucket, download_iterations, download_threads, output_dir):
+def download(endpoint, access_key, secret_key, bucket, num_files, file_size, download_iterations, download_threads, output_dir):
     print(f'📥 Download test')
 
     try:
-        print(f'Spawning threads... [download_threads: {download_threads}]')
-        threads = []
-        for i in range(download_threads):
-            thread = DownloadIteration(i+1, endpoint=endpoint, access_key=access_key, secret_key=secret_key, bucket=bucket, download_iterations=download_iterations)
-            threads.append(thread)
-            thread.start()
-
-        for thread in threads:
-            thread.join()
-
-        for thread in threads:
-            print(thread.get_stats())
-
+        download_start_time = datetime.datetime.now()
+
+        timestamp = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.000%fZ')
+        output_filename = f'download-report-{timestamp}.csv'
+        output_filepath = output_dir + '/' + output_filename
+
+        with open(output_filepath, 'w', buffering=1) as csvfile:
+            csvfile.write(f'threadid,iteration_no,object_name,object_size_bytes,object_elapsed_time_seconds,error\n')
+            csvfile_lock = Lock()
+
+            print(f'Downloading files... [CSV report: {output_filepath}]')
+            print(f'Spawning threads... [download_threads: {download_threads}]')
+            threads = []
+            for i in range(download_threads):
+                threadid = i+1
+                thread = DownloadIteration(threadid=threadid, endpoint=endpoint, access_key=access_key, secret_key=secret_key, bucket=bucket,
+                                           num_files=num_files, file_size=file_size, download_iterations=download_iterations,
+                                           csvfile=csvfile, csvfile_lock=csvfile_lock)
+                threads.append(thread)
+                thread.start()
+
+            for thread in threads:
+                thread.join()
+
+        output_filesize = os.path.getsize(output_filepath)
+        print(f'CSV report generated! [{output_filepath}] ({output_filesize} bytes)')
+
+        download_end_time = datetime.datetime.now()
+        download_elapsed_time_seconds = (download_end_time - download_start_time).total_seconds()
+        print(f'Download finished! [{download_elapsed_time_seconds} seconds]')
     except Exception as e:
         print(f'Exception: {e}')
 
 
 def main(endpoint:str, access_key=str, secret_key=str, bucket=str, num_files=int, file_size=int,
-         only_upload=bool, only_download=bool, download_iterations=int, download_threads=int, output_dir=str):
+         download_iterations=int, download_threads=int, output_dir=str, only_upload=bool, only_download=bool):
```

Comment:
all arguments should be required, I think instead of `=` it should be `:`, e.g. instead of `only_upload=bool` it should be `only_upload:bool`